ADAL/src/ADAuthenticationErrorConverter.m

```diff
@@ -43,30 +43,42 @@ @implementation ADAuthenticationErrorConverter
-                                   @(MSIDErrorInvalidDeveloperParameter):@(AD_ERROR_DEVELOPER_INVALID_ARGUMENT),
-                                   @(MSIDErrorAmbiguousAuthority): @(AD_ERROR_UNEXPECTED),
-                                   @(MSIDErrorInteractionRequired):@(AD_ERROR_SERVER_USER_INPUT_NEEDED),
-                                   @(MSIDErrorCacheMultipleUsers):@(AD_ERROR_CACHE_MULTIPLE_USERS),
-                                   @(MSIDErrorTokenCacheItemFailure):@(AD_ERROR_CACHE_BAD_FORMAT),
-                                   @(MSIDErrorWrapperCacheFailure): @(AD_ERROR_CACHE_BAD_FORMAT),
-                                   @(MSIDErrorCacheBadFormat): @(AD_ERROR_CACHE_BAD_FORMAT),
-                                   @(MSIDErrorCacheVersionMismatch): @(AD_ERROR_CACHE_VERSION_MISMATCH),
-                                   @(MSIDErrorServerInvalidResponse):@(AD_ERROR_SERVER_INVALID_RESPONSE),
-                                   @(MSIDErrorServerRefreshTokenRejected):@(AD_ERROR_SERVER_REFRESH_TOKEN_REJECTED),
-                                   @(MSIDErrorServerOauth): @(AD_ERROR_SERVER_OAUTH),
-                                   @(MSIDErrorInvalidRequest): @(AD_ERROR_SERVER_OAUTH),
-                                   @(MSIDErrorInvalidClient): @(AD_ERROR_SERVER_OAUTH),
-                                   @(MSIDErrorInvalidGrant): @(AD_ERROR_SERVER_REFRESH_TOKEN_REJECTED),
-                                   @(MSIDErrorInvalidParameter):@(AD_ERROR_DEVELOPER_INVALID_ARGUMENT),
-                                   @(MSIDErrorDeveloperAuthorityValidation):@(AD_ERROR_DEVELOPER_AUTHORITY_VALIDATION),
-                                   @(MSIDErrorUnsupportedFunctionality): @(AD_ERROR_UNEXPECTED)
+                                   // General
+                                   @(MSIDErrorInternal) : @(AD_ERROR_UNEXPECTED),
+                                   @(MSIDErrorInvalidInternalParameter) : @(AD_ERROR_UNEXPECTED),
+                                   @(MSIDErrorInvalidDeveloperParameter) :@(AD_ERROR_DEVELOPER_INVALID_ARGUMENT),
+                                   @(MSIDErrorUnsupportedFunctionality): @(AD_ERROR_UNEXPECTED),
+                                   // Cache
+                                   @(MSIDErrorCacheMultipleUsers) : @(AD_ERROR_CACHE_MULTIPLE_USERS),
+                                   @(MSIDErrorCacheBadFormat) : @(AD_ERROR_CACHE_BAD_FORMAT),
+                                   // Authority Validation
+                                   @(MSIDErrorAuthorityValidation) : @(AD_ERROR_DEVELOPER_AUTHORITY_VALIDATION),
+                                   // Interactive flow
+                                   @(MSIDErrorAuthorizationFailed) : @(AD_ERROR_SERVER_AUTHORIZATION_CODE),
+                                   @(MSIDErrorUserCancel) : @(AD_ERROR_UI_USER_CANCEL),
+                                   @(MSIDErrorSessionCanceledProgramatically) : @(AD_ERROR_UNEXPECTED),
```

Comment:
This currently produces AD_ERROR_UI_USER_CANCEL in ADAL (not very logical, but let's not break it), please change :)

Reply:
Done